### ISSUE TITLE
Systematically clean up exercise documentation?

### ISSUE THREAD

**Issue by kytrinyx** [exercism/discussions#42]
We have a lot of different places that we put information that eventually ends up in the exercise README.
1. x-common: exercise description (md) and metadata (yml) - generic information about problem
2. $TRACK/SETUP.md - information relevant to all exercises in a given track
3. $TRACK/exercises/$SLUG/HINTS.md - information specific to a given exercise

We also have `$TRACK/docs/TESTS.md` and I recall creating a `GETTING_STARTED.md` file for the hello-world in Ruby.

There are a number of things we're trying to accomplish with these various things.
- On the website itself, we want to explain how to get started in a language (install dependencies, learn enough to get started), as well as the process of working through an exercise (how to run the tests).
- When someone fetches an exercise in a track, they should have enough information in the README to solve the exercise. They shouldn't have to go look up exercism-specific or test-specific documentation. They should only need to go elsewhere to look for help or documentation if they don't know enough of the language to write the code that they want to write.
- We should be able to tell people in a straight-forward way where to look for answers (not "it's in a few different places, try x, y, and z).
- We want to make it as easy as possible to maintain the documentation. Reduce duplicate information.
- We want to avoid confusing people with language-specific descriptions or terms in the generic information. So we shouldn't say that tests are skipped, because not all languages use the term `skip`... and not all tracks skip exercises.

Should we simplify to some extent? Should we put the information about how to run the tests in `$TRACK/docs/TESTS.md`, and remove that from SETUP.md, and include TESTS.md in the exercise README?

Should we stick GETTING_STARTED.md into HINTS.md so that it gets included in the README? Then we don't have to tell people to look in a completely different place than usual, which would be nice.

If we decide to do a cleanup, what are the specific action items that we should create individual issues for? (Individual issues will make it much easier for different people to tackle small things independently of one another, and this has been pretty successful in helping new contributors find some easy wins to start off with).

**Comment by scottwhudson** [comment]
Something like [Gitbook](https://www.gitbook.com/) would be great for this.  It's free for open books, uses Markdown, is easily searchable, renders [code snippets](https://deezaster.gitbooks.io/swift-snippets/content/Classes_Hashable_Equatable.html), handles large amounts of content well, and plays very nicely with Github. 

[Front End Handbook](http://www.frontendhandbook.com/) is a great example of how something like this might look, along with the corresponding [repo](https://github.com/FrontendMasters/front-end-handbook).  

Gitbook looks clean and seems to work well, it might be worth looking into centralizing all documentation into a single repo and rendering it with a tool like this. You'd get searching, reduced duplication, repo cleanup, and a nice UI.

**Comment by kytrinyx** [comment]
Thanks, @scottwhudson!

There are a couple of things that I want to optimize for:

- Documentation specific to an exercise implementation should live with that implementation (otherwise it's likely that people won't see the documentation when working on the exercise)
- Documentation specific to a language track should live in that language track

Keeping language-specific stuff in the repository for that language means that the maintainers on that language track automatically see changes to things and can review, discuss, and merge.

It would be interesting to look into Gitbook for the contributing documentation to exercism. I'll take a look at it with that in mind.

**Comment by jtigger** [comment]
I've been knee-deep in Contributing documentation; I too am looking into Gitbook...

**Comment by kytrinyx** [comment]
We ended up using straight-up markdown in a repository for project-wide documentation (http://github.com/exercism/docs).

We've documented the exercise README stuff here: https://github.com/exercism/docs/blob/master/contributing-to-language-tracks/exercise-readmes.md

We're working on clarifying one last detail: renaming the SETUP.md / TRACK_HINTS.md and deciding on the best location for it (discussion is here: https://github.com/exercism/discussions/issues/147).

As such, I think that we've gotten what we needed. I'm closing this.
